### PR TITLE
Add a `double-click` callback to the `TouchArea`

### PR DESCRIPTION
**Pull request by hunger**

### CODE REVIEW

File: internal/core/items.rs [hunk 1]
```diff
@@ -508,8 +532,12 @@ impl Item for TouchArea {
                     kind: PointerEventKind::Down,
                     modifiers: window_adapter.window().0.modifiers.get().into(),
                 },));
+
+                InputEventResult::GrabMouse
             }
             MouseEvent::Exit => {
+                self.stop_clicked_timer();
```
Comment (ogoffart) [inline internal/core/items.rs:539]: Don't we need to send the click in that case?

**Review by ogoffart — COMMENTED** [review]
One thing we should do is be very explicit in the documentation about what is the behaviour when there is a click handler and when there isn't.

**Comment by ogoffart** [comment]
I think this looks good. We should just document this new callback and describe that assigning a double click handler also change the click behavior.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,13 @@
 # Changelog
 All notable changes to this project are documented in this file.
 
+## [Unreleased]
+
+### Slint Language
+
+ - The `double-clicked` callback in `TouchArea` is triggered when a `TouchArea`
+   is clicked twice in rapid succession.
```
Comment (tronical) [inline CHANGELOG.md:11]:
```suggestion
 - Added a `double-clicked` callback in `TouchArea`, which is triggered when a `TouchArea`
   is clicked twice in rapid succession.
```

File: internal/core/items.rs [hunk 3]
```diff
@@ -424,13 +425,48 @@ pub struct TouchArea {
...
+                }
+            },
+        );
+        assert!(self.clicked_timer.running());
+    }
+
+    fn stop_clicked_timer(self: Pin<&Self>) -> bool {
+        let result = self.clicked_timer.running();
+        self.clicked_timer.stop();
+        assert!(!self.clicked_timer.running());
```
Comment (tronical) [inline internal/core/items.rs:467]:
```suggestion
        debug_assert!(!self.clicked_timer.running());
```

That said, I don't like the idea of asserting here. In the event of this happening the user application crashes, which really doesn't do anyone a favor. If this is supposed to "verify" that our Time behaves, then the right thing is to do that in a unit test (which you did), so I'd remove this TBH and rely on Timer working (it's tested elsewhere, not in production code paths).

File: internal/core/tests.rs [hunk 4]
```diff
@@ -33,7 +33,6 @@ pub extern "C" fn slint_get_mocked_time() -> u64 {
 /// Simulate a click on a position within the component.
 #[no_mangle]
 pub extern "C" fn slint_send_mouse_click(
-    _component: &crate::item_tree::ItemTreeRc,
```
[the remark this answers is not in the document]
Comment (tronical) [reply internal/core/tests.rs:36]: No worries, this isn't public API. We can change it as we like. It's not documented, there are no public header files, after all those functions are emitted into the `cbindgen_private` namespaces.

**Pull request merged**